spec/predicates/deployment.md

```diff
@@ -0,0 +1,477 @@
+(e.g., a service account, a pod ID). For scope types that identify a resource implicitly via an authorization URI (see [Schema](#schema)), the verifier matches the value against the URI in the configuration (See [Configuration](#configuration)). Non-empty fields add constraints to the protection scope and are _always_ interpreted as a logical "AND". The verifier MUST compare each scope value to its expected value using an equality comparison. If the values are all equal, verification passes. Otherwise, it MUST fail. Unset scopes (either a scope type with an empty value or a non-present scope) are interpreted as "any value" and are ignored.
+
+### Supported Scopes
+
+#### Kubernetes pod scope
+
+The specifications defines the Kubernetes's pod scope as follows:
+
+```shell
+kubernetes.io/pod/service_account/v1  string: A k8 service account
+kubernetes.io/pod/cluster_id/v1       string: A cluster ID
+kubernetes.io/pod/namespace/          string: A namespace
+kubernetes.io/pod/cluster_name/v1     string: A cluster name
+```
+
+The scope match verification compares each non-empty scope value against the corresponding
+environment value the artifact is to be deployed:
+
+```shell
+attestation's "kubernetes.io/pod/service_account" == environment's "k8's service_account" AND
+attestation's "kubernetes.io/pod/cluster_id" == environment's "k8'scluster_id" AND 
+...
+```
+
+#### GCP scope
+
+The specifications defines the GCP scope as follows:
+
+```shell
+cloud.google.com/service_account/v1 string: A GCP service account
+cloud.google.com/location/v1        string: A location
+cloud.google.com/project_id/v1      string: A project id
+```
+
+The scope match verification compares each non-empty scope value against the corresponding
+environment value the artifact is to be deployed:
+
+
+```shell
+attestation's "cloud.google.com/service_account" == environment's "GCP service_account" AND
+attestation's "cloud.google.com/project_id" == environment's "GCP project ID" AND 
+...
+```
+
+#### Spiffe
+
+The specifications define the Spiffe scope as follows:
+
+```shell
+spiffe.io/id/v1 string: The Spiffe ID
+```
+
+The scope match verification compares each non-empty scope value against the corresponding
+environment value the artifact is to be deployed:
+
+
+```shell
+attestation's "spiffe.io/id" == environment's "Spiffe ID" AND
+...
+```
+
+#### Custom scopes
+
+One can define their own scopes. To avoid scope type name collisions, the scope type name MUST be a unique URI, such as:
+
+```shell
+my.myproject.com/resource/v1   string: resource for environment my.myproject.com
+```
+
+If a scope type is unrecognized or not supported by the verifier, verification MUST fail.
+
+## Examples
+
+### Example 1: Single scope
+
+**Trusted (single) root configuration**:
```

Comment:
What are your thoughts here?

(changes are optional)

Reply:
I'm totally OK making these changes if it helps the reader. I think I agree with you it will help. I'll ping here when I've made the changes. Thanks again for the suggestions.